### ISSUE TITLE
[BUG] have_table matcher with with_cols: option behaves unexpectedly

### ISSUE THREAD

**Issue by Looooong** [teamcapybara/capybara#2287]
## Meta
Capybara Version: master branch, latest commit 36c2d2c1a5dff8a1a4cd586f5024e1dbda77a8bb
Driver Information (and browser if relevant): RackTest

## Steps to reproduce
I added the following examples to [lib/capybara/spec/session/has_table_spec.rb](https://github.com/teamcapybara/capybara/blob/master/lib/capybara/spec/session/has_table_spec.rb):
```ruby
  it 'should not match (fail)' do
    expect(@session).to have_no_table('Vertical Headers', with_cols:
      [
        {
          'First Name' => 'What?',
          'What?' => 'Walpole',
          'City' => 'Oceanside' # This line makes the example fail
        }
      ]
    )
  end

  it 'should not match (pass)' do
    expect(@session).to have_no_table('Vertical Headers', with_cols:
      [
        {
          'First Name' => 'What?',
          'What?' => 'Walpole',
          'City' => 'What?'
        }
      ]
    )
  end

  it 'should not match (pass)' do
    expect(@session).to have_no_table('Vertical Headers', with_cols:
      [
        {
          'First Name' => 'What?',
          'What?' => 'Walpole'
        }
      ]
    )
  end
```

## Expected Behavior
All specs should pass.

## Actual Behavior
The first spec fails.

```
  1) Capybara::Session selenium_chrome #has_table? should not match (fail)
     Failure/Error:
       expect(@session).to have_no_table('Vertical Headers', with_cols:
         [
           {
             'First Name' => 'What?',
             'What?' => 'Walpole',
             'City' => 'Oceanside'
           }
         ]
       )
     
       expected not to find visible table "Vertical Headers", found 1 match: "Vertical Headers\nFirst Name Thomas Danilo Vern Ratke Palmer\nLast Name Walpole Wilkinson Konopelski Lawrence Sawayn\nCity Oceanside Johnsonville Everette East Sorayashire West Trinidad"
     # ./lib/capybara/spec/session/has_table_spec.rb:67:in `block (2 levels) in <top (required)>'
```

**Comment by twalpole** [comment]
I'll look at this over the weekend if no one else fixes it before

**Comment by twalpole** [comment]
I think the issue is probably that `prev_col_position?` in table.rb should be something like

```
def prev_col_position?(cell)
   cell & XPath.position.equals(cell_position(cell))
 end


but I need more time than I currently have to fully confirm that

**Comment by Looooong** [comment]
```ruby
  def prev_col_position?(cell)
    cell & XPath.position.equals(cell_position(cell))
  end
```

I tried your solution and the specs passed.

**Comment by twalpole** [comment]
Fixed in 3.30
